internal/website/docs/next/guides/sitemaps.mdx

```diff
@@ -0,0 +1,232 @@
+
+### Creating a `/robots.txt` Route
+
+There will most likely be instances where you will need to create a `/robots.txt` route. This is a file that tells search engines what to do with your site. We have a handy function that you can define in your `handleSitemapRequests` config object that has an argument of `sitemapUrl` to be used:
+
+```tsx title="src/pages/_middleware.ts" {8-15}
+import { handleSitemapRequests } from '@faustjs/next';
+import { NextRequest, NextResponse } from 'next/server';
+
+export default async function _middleware(req: NextRequest) {
+  const sitemapRequest = await handleSitemapRequests(req, {
+    wpUrl: process.env.NEXT_PUBLIC_WORDPRESS_URL,
+    sitemapIndexPath: `/wp-sitemap.xml`,
+    async robotsTxt(sitemapUrl) {
+      return `
+        User-agent: *
+        Allow: /
+
+        Sitemap: ${sitemapUrl}
+      `;
+    },
+  });
+
+  if (sitemapRequest) {
+    return sitemapRequest;
+  }
+
+  return NextResponse.next();
+}
+```
+
+## Examples
+
+Below you can find some drop in examples for different XML Sitemap configurations on WordPress.
+
+:::note
+
+Be sure to define your `pages` property in your `handleSitemapRequests` config object if you have any Next.js specific pages you want included in your sitemap.
+
+:::
+
+### Usage with default WordPress Sitemaps
+
+Below is a drop in configuration using default WordPress sitemaps. This assumes you want to ignore the "users" sitemap:
+
+```tsx title="src/pages/_middleware.ts"
+import { handleSitemapRequests } from '@faustjs/next';
+import { NextRequest, NextResponse } from 'next/server';
+
+export default async function _middleware(req: NextRequest) {
+  const sitemapRequest = await handleSitemapRequests(req, {
+    wpUrl: process.env.NEXT_PUBLIC_WORDPRESS_URL,
+    sitemapIndexPath: `/wp-sitemap.xml`,
+    sitemapPathsToIgnore: ['/wp-sitemap-users-*'],
+    async robotsTxt(sitemapUrl) {
+      return `
+        User-agent: *
+        Allow: /
+
+        Sitemap: ${sitemapUrl}
+      `;
+    },
+  });
+
+  if (sitemapRequest) {
+    return sitemapRequest;
+  }
+
+  return NextResponse.next();
+}
+```
+
+### Usage with Yoast
+
+Below is a drop in configuration using Yoast's sitemap. This assumes you want to ignore the "author" sitemap:
+
+```tsx title="src/pages/_middleware.ts"
+import { handleSitemapRequests } from '@faustjs/next';
+import { NextRequest, NextResponse } from 'next/server';
+
+export default async function _middleware(req: NextRequest) {
+  const sitemapRequest = await handleSitemapRequests(req, {
+    wpUrl: process.env.NEXT_PUBLIC_WORDPRESS_URL,
+    sitemapIndexPath: `/sitemap_index.xml`,
+    sitemapPathsToIgnore: ['/author-sitemap.xml'],
+    async robotsTxt(sitemapUrl) {
+      return `
+        User-agent: *
+        Allow: /
+
+        Sitemap: ${sitemapUrl}
+      `;
+    },
+  });
+
+  if (sitemapRequest) {
+    return sitemapRequest;
+  }
+
+  return NextResponse.next();
+}
+```
```

Comment:
Note that `/sitemap.xml` redirects to `/wp-sitemap.xml` or if Yoast is activated, `/sitemap_index.html`. Knowing that would we still need to include both sitemap examples?

Reply:
@josephfusco That was not my experience when using the default WordPress sitemap. For me, it only exposed `/wp-sitemap.xml`. Could it have been that you experienced this when permalinks were not refreshed after disabling Yoast?

https://make.wordpress.org/core/2020/07/22/new-xml-sitemaps-functionality-in-wordpress-5-5/

Reply:
Here are my findings with this.

WP install with no plugins,
✅ **/sitemap.xml** successfully redirects to **/wp-sitemap.xml**

If I activate Yoast, & do not flush WP permalinks,
❌ **/sitemap.xml** redirects but shows 404.

If I activate Yoast, & flush WP permalinks,
✅ **/sitemap.xml** successfully redirects to **/sitemap_index.html**

If I deactivate Yoast, & do not flush WP permalinks,
❌ **/sitemap.xml** redirects but shows 404.

If I deactivate Yoast & flush WP permalinks,
✅ **/sitemap.xml** successfully redirects to **/wp-sitemap.xml**

Giving this more thought, I'm good with us being explicit here as I'm not a fan of asking a developer to flush permalinks.